syntax/go.vim

```diff
@@ -301,38 +291,42 @@ hi def link     goOperator          Operator
 
 " Functions;
 if g:go_highlight_functions != 0
-  syn match goFunction              /\(func\s\+\)\@<=\w\+\((\)\@=/
-  syn match goFunction              /\()\s\+\)\@<=\w\+\((\)\@=/
+  syn match goDeclaration       /\<func\>/ nextgroup=goReceiver,goFunction skipwhite skipnl
+  syn match goReceiver /([^),]\+)/ contained nextgroup=goFunction contains=goReceiverType skipwhite skipnl
+  syn match goReceiverType /\(\s\|*\)\w\+)/hs=s+1,he=e-1 contained
+  syn match goFunction /\w\+/ contained
+else
```

Comment:
Why do we have the else clause here? If it's not enabled we should not highlight anything at all